emergence_lib/src/terrain/mod.rs

```diff
@@ -105,22 +108,77 @@ impl TerrainBundle {
     }
 }
 
+/// Generates the merged mesh for all columns in the world.
+///
+/// This needs to be called whenever the height of tiles changes.
+fn generate_mesh_for_columns(map_geometry: &MapGeometry) -> Mesh {
+    let mut mesh_info_vec = Vec::new();
+
+    for tile_pos in map_geometry.valid_tile_positions() {
+        let height = map_geometry.get_height(tile_pos).unwrap().0;
+        let mesh_info = ColumnMeshBuilder::new(&map_geometry.layout, height)
+            .without_bottom_face()
```

Comment:
adding `at(tile_pos)` might be what you want to merge meshes with the correct offset

Reply:
Aha, that's what I was missing!